tutorials/10_DebuggingREMIND.md

```diff
@@ -0,0 +1,169 @@
+
+- the model remains locked? start `vi .lock` in the REMIND folder and carefully delete (with `dd`, then save with `:w`) the lines of older runs that may still be present there because you killed these runs.
+- because of a locked folder in the R libraries, outdated packages were loaded: consider [creating a new snapshot](4_RunningREMINDandMAgPIE.md#create-snapshot-of-r-libraries)
+- input files are missing or outdated? delete `input/source_files.log` or set `cfg$force_download <- TRUE` to force the loading of new ones. If this doesn't help, the input generation may be broken
+- you use an outdated git branch? check `git log` or use main branch with `git checkout main/develop; git pull`
+- a recent change in `prepare_and_run` may have broken the prepare scripts, check the [file history](https://github.com/remindmodel/remind/commits/develop/scripts/start/prepare_and_run.R) for changes.
+
+Case 2: REMIND did produce an error
+-----------------------------------
+
+In this case, after "REMIND run finished!", you find a short "Model Summary", where the script looks for some files in the output subfolder and gives you its interpretation which is helpful for knowing were to look next:
+```
+  gams_runtime is 10.6 hours.
+  full.gms exists, so the REMIND GAMS code was generated.
+  full.log and full.lst exist, so GAMS did run.
+! abort.gdx exists, containing the latest GAMS data including changes after the last solve statement.
+  fulldata.gdx exists, so at least one iteration was successful.
+  Number of iterations: 14
+  Modelstat: 4
+! full.log states: *** Status: Execution error(s)
+```
+
+If gams_runtime was short (say: up to 10 minutes), this implies an error right at the beginning of the run.
+If full.gms does not exist, it is an error in the preparation script, not in GAMS.
+
+If full.log exists, this is the next place to look at. Either open it in your favorite text editor or use the console, change directory to the output subfolder and type:
+```bash
+vi full.log
+```
+and then type `G` to get to the end of the file. `:q` finishes looking at the file.
+
+You may find:
+
+### Case 2a: compilation errors
+
+These are errors in the GAMS code. Run `gamscompile` in the REMIND folder to get more information and try to identify incorrect GAMS code.
+Tutorials such as as the [McCarl GAMS User Guide](https://www.gams.com/mccarlGuide/) may be helpful.
+
+### Case 2b: execution errors
+
+Have the next look at
+
+```
+vi full.lst
+```
+Hit `G` to go to the end of the file and search backwards for errors whose lines start with four asterisks by typing:
+```
+?^\*\*\*\*
+```
+(If you add `map f /*\*\*\*<CR>` to your `/home/username/.vimrc` file, you can simply press `f`).
```

Comment:
maybe add a note that some functionality that no one ever uses (and those who know it will have better ways to do the four asterisk search anyway) will be replaced by a super useful one `:)`

Reply:
I personally think it is helpful, I think it comes from the VI tutorial on redmine.

Reply:
I added the info that this overwrites the normal single-character search, so people can decide themselves